### ISSUE TITLE
Crashes when used with flipper.

### ISSUE THREAD

**Issue by dkahdwk** [AgoraIO-Extensions/react-native-agora#648]
**Describe the bug**

After installing react-native-agora, if you run the app in the ios simulator, it will crash with the flipper and force the app to shut down.

In a new project, I found that running the flipper crashed, but I found that it crashed with the react-native-agora library in the bug report.

After deleting the react-native-agora library, I checked that it works fine when I run the flipper and run the app. (It works normally in the xcode build with the flipper running, but it crashes when you run the app in the terminal.)

Below is the Flipper crash report.

```jsx
Thread 13 Crashed:: com.facebook.SocketRocket.NetworkThread
0   <translation info unavailable>	       0x107246de8 ???
1   <translation info unavailable>	       0x107247244 ???
2   AgoraDrmLoaderExtension       	       0x10f0fa479 0x10eef0000 + 2139257
3   AgoraDrmLoaderExtension       	       0x10ef9b405 d2i_PrivateKey + 229
4   AgoraDrmLoaderExtension       	       0x10f0122b1 PEM_read_bio_PrivateKey + 625
5   soldoc                        	       0x10053cde7 facebook::flipper::generateCertPKCS12(char const*, char const*, char const*, char const*, char const*, char const*) + 215 (CertificateUtils.cpp:304)
6   soldoc                        	       0x10053ef66 facebook::flipper::ConnectionContextStore::getCertificate() + 454 (ConnectionContextStore.cpp:197)
7   soldoc                        	       0x10077094a facebook::flipper::FlipperWebSocket::connect(facebook::flipper::FlipperConnectionManager*)::$_0::operator()(char*, unsigned long) const + 58 (FlipperWebSocket.mm:117)
8   soldoc                        	       0x1007708e9 decltype(static_cast<facebook::flipper::FlipperWebSocket::connect(facebook::flipper::FlipperConnectionManager*)::$_0&>(fp)(static_cast<char*>(fp0), static_cast<unsigned long>(fp0))) std::__1::__invoke<facebook::flipper::FlipperWebSocket::connect(facebook::flipper::FlipperConnectionManager*)::$_0&, char*, unsigned long>(facebook::flipper::FlipperWebSocket::connect(facebook::flipper::FlipperConnectionManager*)::$_0&, char*&&, unsigned long&&) + 57 (type_traits:3918)
9   soldoc                        	       0x10077087a std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > std::__1::__invoke_void_return_wrapper<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, false>::__call<facebook::flipper::FlipperWebSocket::connect(facebook::flipper::FlipperConnectionManager*)::$_0&, char*, unsigned long>(facebook::flipper::FlipperWebSocket::connect(facebook::flipper::FlipperConnectionManager*)::$_0&, char*&&, unsigned long&&) + 90 (invoke.h:30)
10  soldoc                        	       0x1007707ea std::__1::__function::__alloc_func<facebook::flipper::FlipperWebSocket::connect(facebook::flipper::FlipperConnectionManager*)::$_0, std::__1::allocator<facebook::flipper::FlipperWebSocket::connect(facebook::flipper::FlipperConnectionManager*)::$_0>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (char*, unsigned long)>::operator()(char*&&, unsigned long&&) + 90 (function.h:178)
11  soldoc                        	       0x10076f4c9 std::__1::__function::__func<facebook::flipper::FlipperWebSocket::connect(facebook::flipper::FlipperConnectionManager*)::$_0, std::__1::allocator<facebook::flipper::FlipperWebSocket::connect(facebook::flipper::FlipperConnectionManager*)::$_0>, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (char*, unsigned long)>::operator()(char*&&, unsigned long&&) + 89 (function.h:352)
12  soldoc                        	       0x100766442 std::__1::__function::__value_func<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (char*, unsigned long)>::operator()(char*&&, unsigned long&&) const + 114 (function.h:505)
13  soldoc                        	       0x100763de5 std::__1::function<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > (char*, unsigned long)>::operator()(char*, unsigned long) const + 85 (function.h:1182)
14  soldoc                        	       0x1007636fa -[FlipperClientCertificateSecurityPolicy updateSecurityOptionsInStream:] + 458 (FlipperPlatformWebSocket.mm:58)
15  soldoc                        	       0x1010a5d0f -[SRWebSocket _updateSecureStreamOptions] + 111 (SRWebSocket.m:472)
16  soldoc                        	       0x1010a5247 -[SRWebSocket _connectionDoneWithError:readStream:writeStream:] + 247 (SRWebSocket.m:348)
17  soldoc                        	       0x1010a5114 __19-[SRWebSocket open]_block_invoke.66 + 132 (SRWebSocket.m:334)
18  soldoc                        	       0x1010a08db -[SRProxyConnect _didConnect] + 587 (SRProxyConnect.m:117)
19  soldoc                        	       0x1010a21a8 -[SRProxyConnect stream:handleEvent:] + 232 (SRProxyConnect.m:347)
20  CoreFoundation                	       0x11e5e468c _signalEventSync + 219
21  CoreFoundation                	       0x11e5e1d44 _cfstream_solo_signalEventSync + 266
22  CoreFoundation                	       0x11e5e1bfc _CFStreamSignalEvent + 447
23  CFNetwork                     	       0x110887c9a 0x110785000 + 1059994
24  CFNetwork                     	       0x11088f66b 0x110785000 + 1091179
25  CoreFoundation                	       0x11e5d44d8 __CFSocketPerformV0 + 652
26  CoreFoundation                	       0x11e5c0fe5 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
27  CoreFoundation                	       0x11e5c0f24 __CFRunLoopDoSource0 + 157
28  CoreFoundation                	       0x11e5c0721 __CFRunLoopDoSources0 + 212
29  CoreFoundation                	       0x11e5bae23 __CFRunLoopRun + 927
30  CoreFoundation                	       0x11e5ba6a7 CFRunLoopRunSpecific + 560
31  Foundation                    	       0x1140ef8b4 -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 213
32  soldoc                        	       0x1010a31bb -[SRRunLoopThread main] + 267 (SRRunLoopThread.m:70)
33  Foundation                    	       0x11411911b __NSThread__start__ + 1009
34  libsystem_pthread.dylib       	       0x11eca24e1 _pthread_start + 125
35  libsystem_pthread.dylib       	       0x11ec9df6b thread_start + 15
```

**To Reproduce**
1. install react-native-agora
2. open flipper
3. npm run ios
4. crash!

**Expected behavior**

**Screenshots**

**Smartphone (please complete the following information):**
 - iPhone 14
 - OS: iOS 16.2
 - react-native 0.71.+ or react-native 0.68.9

**Additional context**

**Comment by wuxushun** [comment]
same issues, i hope support Fliper. it so important

**Comment by LichKing-2234** [comment]
fixed at 4.2.0
